waltz-tools/src/main/java/com/wepay/waltz/tools/performance/PerformanceCli.java

```diff
@@ -422,21 +445,30 @@ protected void processCmd(CommandLine cmd) throws SubCommandFailedException {
                         throw new IllegalArgumentException("Number of active partitions must be greater of equals to 1");
                     }
                 }
+                Map<Integer, Long> partitionHighWaterMarkMap = cmd.hasOption("up_to_date_watermark")
+                        ? getHighWaterMarkPerPartition(waltzClientConfig, numActivePartitions)
+                        : new HashMap<>();
 
                 // produce transactions to consume
                 // since we only care about consumer performance, we can create
                 // transactions with multiple producer to expedite the test
                 int numThread = numTxn > DEFAULT_NUM_PRODUCERS ? DEFAULT_NUM_PRODUCERS : numTxn;
                 int txnPerThread = numTxn / numThread;
+                int leftOutTransactions = numTxn % numThread;
                 ExecutorService executor = Executors.newFixedThreadPool(numThread);
                 for (int i = 0; i < numThread; i++) {
-                    ProducerCallbacks callbacks = new ProducerCallbacks();
+                    ProducerCallbacks callbacks = new ProducerCallbacks(partitionHighWaterMarkMap);
                     WaltzClient client = new WaltzClient(callbacks, waltzClientConfig);
-                    executor.execute(new ProducerThread(txnPerThread, client));
+                    if (leftOutTransactions > 0) {
+                        executor.execute(new ProducerThread(txnPerThread + 1, client));
+                        leftOutTransactions--;
+                    } else {
+                        executor.execute(new ProducerThread(txnPerThread, client));
+                    }
```

Comment:
One more way instead of if-else can be:
```
leftOutTransactions-- ? executor.execute(new ProducerThread(txnPerThread + 1, client)) : 
executor.execute(new ProducerThread(txnPerThread, client))```

Reply:
my bad!!! completely forgot that `leftOutTransactions--` is a C thing and not Java...lets just leave it as if else loop.